//
// Copyright (C) 2024 EA group inc.
// Author: Jeff.li lijippy@163.com
// All rights reserved.
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published
// by the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
//


syntax="proto2";

package melon;
option java_package = "com.melon";
option java_outer_classname="StreamingRpcProto";

message StreamSettings {
    required int64 stream_id = 1;
    optional bool need_feedback = 2 [default = false];
    optional bool writable = 3 [default = false];
}

enum FrameType {
    FRAME_TYPE_UNKNOWN = 0;
    FRAME_TYPE_RST = 1;
    FRAME_TYPE_CLOSE = 2;
    FRAME_TYPE_DATA = 3;
    FRAME_TYPE_FEEDBACK= 4;
}

message StreamFrameMeta {
    required int64 stream_id = 1;
    optional int64 source_stream_id = 2;
    optional FrameType frame_type = 3;
    optional bool has_continuation = 4;
    optional Feedback feedback = 5;
}

message Feedback {
    optional int64 consumed_size = 1;
}
